homeassistant/components/mqtt/models.py

```diff
@@ -51,6 +51,22 @@ class PayloadSentinel(StrEnum):
 type PublishPayloadType = str | bytes | int | float | None
 
 
+def convert_outgoing_mqtt_payload(
+    payload: PublishPayloadType,
+) -> PublishPayloadType:
+    """Ensure correct raw MQTT payload is passed as bytes for publishing."""
+    if isinstance(payload, str) and payload.startswith(("b'", 'b"')):
```

Comment:
Can we assert in a test we don't call `literal_eval` when the string does not start with `b'` or `b"`?

Reply:
Test was expanded